pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -137,10 +137,7 @@ def _getDefaultOptions() -> Dict[str, Any]:
             "Total character workspace": [int, None],
             "Total integer workspace": [int, None],
             "Total real workspace": [int, None],
-            "Save major iteration variables": [
-                list,
-                ["step", "merit", "feasibility", "optimality", "penalty"],
-            ],
+            "Save major iteration variables": [list, []],
```

Comment:
I think it would be better to add to the default list than have variables that cannot be turned off. This would make the code easier to understand, and it would be consistent with how we handle saving variables in ADflow.

Reply:
I also debated about this. I think the reason I preferred this is because, if I want to add another one then I have to copy/paste the list of defaults first from somewhere (probably docs) which is a little annoying. Also, given the small storage requirement I don't think it's necessary to allow users to remove the default ones. Happy to change things though.

Reply:
What do you think @lamkina?

Reply:
I tend to agree with @nwu63 in this case.  The storage requirement for the default variables is pretty insignificant and from an OptView perspective it's nice to have an expected minimal set of data for every SNOPT case.

From the user perspective, I'm always against extra steps to get simple cases working.  If the proposed alternative means users need to read the docs to understand the default usage of recoring, I think that is too complicated for the intended impact of these changes.

Reply:
Alright, that fine with me. I would like the defaults to be documented though because the empty list suggests nothing is saved by default.

Reply:
That's a good compromise imo.